### PR TITLE
return invalid planeCrossing to reference points with zero B-field in TrackKinematicStatePropagator

### PR DESCRIPTION
**Pull request by slava77**
as discussed in #36788, the `TrackKinematicStatePropagator` can produce a "valid" state with `NaN` values for a propagation to a point with zero magnetic field.

This PR creates a properly invalid (already detectable as bad) result.

@swagata87

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by smuzaffar** [comment]
please test workflow 134.802,136.739,1001.2,50208.0 for el9_aarch64_gcc11

**Comment by smuzaffar** [comment]
failing relvals (e.g. 134.802,136.739,1001.2,50208.0) for `el9_aarch64_gcc11` now work

**Comment by jpata** [comment]
please test workflow 134.802,136.739,1001.2,50208.0 for el9_aarch64_gcc11

**Comment by smuzaffar** [comment]
@jpata , do you want to re-run the same tests for which results are available here https://github.com/cms-sw/cmssw/pull/39183#issuecomment-1227180595

**Comment by jpata** [comment]
I had interpreted the comments as something been broken in those tests (and the bot message was not the usual one). But if everything was as expected, then let's cancel the test.

**Comment by smuzaffar** [comment]
ah sorry, I have updated the [comment](https://github.com/cms-sw/cmssw/pull/39183#issuecomment-1227203648) .
About the bot message, we do not run DQM comparison jobs for aarch64/ppc64le as they take too much time (due to a lot DQM differences between amd64 baseline and non-amd64 PR relvals).

File: RecoVertex/KinematicFitPrimitives/src/TrackKinematicStatePropagator.cc [hunk 1]
```diff
@@ -58,6 +63,8 @@ bool TrackKinematicStatePropagator::willPropagateToTheTransversePCA(const Kinema
   // copied from below...
   FreeTrajectoryState const& fState = state.freeTrajectoryState();
   std::pair<HelixBarrelPlaneCrossingByCircle, BoundPlane::BoundPlanePointer> cros = planeCrossing(fState, point);
+  if (cros.second == nullptr)
```
Comment (mandrenguyen) [inline RecoVertex/KinematicFitPrimitives/src/TrackKinematicStatePropagator.cc:66]: Is this return statement and the one below on L88 both supposed to be active at the same time?  Running on the 100 event test file provided by Shazhad, it seems that only the one below on L88 is executed (twice), but this one is never reached.

Comment (slava77) [reply RecoVertex/KinematicFitPrimitives/src/TrackKinematicStatePropagator.cc:66]: this method `willPropagateToTheTransversePCA` is used to check if the propagation is possible. I can see how someone may want to call it separately from the actual propagation in `propagateToTheTransversePCACharged`.

I guess in the tested setup this part is never called.

Comment (slava77) [reply RecoVertex/KinematicFitPrimitives/src/TrackKinematicStatePropagator.cc:66]: the `willPropagate*` is used in `KinematicParticleVertexFitter`; here the call traces go through `KinematicConstrainedVertexFitterT`.

**Comment by mandrenguyen** [comment]
+reconstruction
Fixes an occasional NaN by sending back any empty KinematicState
Difference to reco are negligible

**Comment by perrotta** [comment]
+1

**Pull request merged**